### ISSUE TITLE
allow new_without_default is broken

### ISSUE THREAD

**Issue by gnzlbg** [rust-lang/rust-clippy#3199]
[Playground](https://play.rust-lang.org/?gist=bd1416a9feed3704b42512407ea8307a&version=stable&mode=debug&edition=2015):

```rust
pub struct Foo {
    pub x: *const i32,
}

impl Foo {
    #[cfg_attr(feature = "cargo-clippy", allow(clippy::new_without_default))]
    pub fn new() -> Self {
        Self {
            x: std::ptr::null(),
        }
    }
}
```

produces

```
warning: you should consider adding a `Default` implementation for `Foo`
  --> src/main.rs:9:5
   |
9  | /     pub fn new() -> Self {
10 | |         Self {
11 | |             x: std::ptr::null(),
12 | |         }
13 | |     }
   | |_____^
   |
   = note: #[warn(clippy::new_without_default)] on by default
   = help: for further information visit https://rust-lang-nursery.github.io/rust-clippy/v0.0.212/index.html#new_without_default
help: try this [...]
```

**Comment by flip1995** [comment]
You have to add the`#[cfg_attr(feature = "cargo-clippy", allow(clippy::new_without_default))]` to the `impl`, not the `new` function. Did this work before?

I think this is expected behavior, so I'm closing this issue. If you think this is unexpected, feel free to reopen.

**Comment by gnzlbg** [comment]
Adding the attribute to the function should raise a warning / error then.

**Comment by gnzlbg** [comment]
Also, the error message for `new_without_default` points to the `new` method, but if the attribute should be applied to the impl, shouldn't it point to the `impl` ? or at least suggest adding the allow on the `impl` instead of the method ?

**Comment by flip1995** [comment]
That is a good point!
